File: softphone-ios/MyLineSoftphone/Models/SipConfig.swift
import Foundation

/// SIP account configuration. Mirrors Android `SipConfig.kt`.
struct SipConfig: Codable, Equatable {
    var server: String = ""
    var port: Int = 5060
    var localPort: Int = 5060
    var rport: Bool = true
    var username: String = ""
    var password: String = ""
    var displayName: String = ""
    var transport: String = "UDP"
    var enabled: Bool = true

    var isValid: Bool {
        !server.isEmpty && !username.isEmpty && !password.isEmpty
    }

    /// Full SIP domain — if the server lacks a dot, append the default realm.
    var domain: String {
        server.contains(".") ? server : "\(server).myline.tel"
    }
}

File: softphone-android/app/src/main/java/com/mylinetelecom/softphone/models/SipConfig.kt
package com.mylinetelecom.softphone.models

data class SipConfig(
    val server: String = "",
    val port: Int = 5060,
    val localPort: Int = 5060,
    val rport: Boolean = true,
    val username: String = "",
    val password: String = "",
    val displayName: String = "",
    val transport: String = "UDP",
    val enabled: Boolean = true
) {
    val isValid: Boolean
        get() = server.isNotBlank() && username.isNotBlank() && password.isNotBlank()

    val domain: String
        get() = if (server.contains(".")) server else "$server.myline.tel"
}
